Textor/Controller/FontPickerViewController.swift

```diff
@@ -17,16 +17,24 @@ class FontPickerViewController: UITableViewController {
 		return searchController
 	}()
 	
-	let fonts: [String] = {
-		var allFonts = [String]()
-		let fontFamilys = UIFont.familyNames
-		for fontFamily in fontFamilys {
-			allFonts += UIFont.fontNames(forFamilyName: fontFamily)
+	let fonts: [Character: [String]] = Dictionary(grouping: UIFont.familyNames.flatMap(UIFont.fontNames(forFamilyName:)).sorted()) { font  in
+		if let character = font.uppercased().first,
+			let characterScalar = "\(character)".unicodeScalars.first,
+			CharacterSet.letters.contains(characterScalar) {
+			return character
+		} else {
+			// Fallback case, in case some installs a font that doesn't start
+			// with an alphabetical character
+			return "*" as Character
 		}
-		
-		return allFonts.sorted()
-	}()
-	var filteredFonts = [String]()
+	}
+	var filteredFonts = [Character: [String]]()
+	// Cache for getting a character for a given index, since ordinarily this
+	// would be O(n) on a String
+	var charactersForIndex: [Int: Character] = [:]
+	
+	var defaultSeparatorColor: UIColor!
```

Comment:
By the way, this code is duplicated across both the font picker as well as the settings view controller. If you think this needs to be fixed please let me know.